samples/connectivity/bluetooth/companion/src/main/java/com/example/platform/connectivity/bluetooth/ble/CompanionDeviceManagerSample.kt

```diff
@@ -0,0 +1,304 @@
+import android.content.Intent
+import android.content.IntentSender
+import android.os.Build
+import android.os.ParcelUuid
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.IntentSenderRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.RequiresApi
+import androidx.compose.animation.AnimatedContent
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Button
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.core.content.getSystemService
+import com.example.platform.base.PermissionBox
+import com.google.android.catalog.framework.annotations.Sample
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executor
+
+@Sample(
+    name = "Companion Device Manager Sample",
+    description = "This samples shows how to use the CDM to pair and connect with BLE devices",
+    documentation = "https://developer.android.com/guide/topics/connectivity/companion-device-pairing",
+    tags = ["bluetooth"],
+)
+@SuppressLint("InlinedApi", "MissingPermission")
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun CompanionDeviceManagerSample() {
+    val context = LocalContext.current
+    val deviceManager = context.getSystemService<CompanionDeviceManager>()
+    val adapter = context.getSystemService<BluetoothManager>()?.adapter
+    var selectedDevice by remember {
+        mutableStateOf<BluetoothDevice?>(null)
+    }
+    if (deviceManager == null || adapter == null) {
+        Text(text = "No Companion device manager found. The device does not support it.")
+    } else {
+        if (selectedDevice == null) {
+            CDMScreen(deviceManager) {
+                selectedDevice = it.device ?: adapter.getRemoteDevice(it.name)
+            }
+        } else {
+            PermissionBox(permission = Manifest.permission.BLUETOOTH_CONNECT) {
+                ConnectDeviceScreen(device = selectedDevice!!) {
+                    selectedDevice = null
+                }
+            }
+        }
+    }
+}
+
+data class AssociatedDevice(
+    val id: Int,
+    val address: String,
+    val name: String,
+    val device: BluetoothDevice?,
+)
```

Comment:
I'm not sure if it makes sense to create a custom AssociatedDevice with a BluetoothDevice field, since the AssociatedDevice provided by CDM already have all the info and it could also return a different type e.g. Wifi or Ble ScanResult. We would recommend the app dev to use as it is unless they definitely know it's a Classic Bluetooth device. In that case, we might want to add a comment here saying the app dev knows it's a Classic BT device.

Reply:
This is to handle the backwards compatibility since depending on the version we get the scan result or the association info but I want to avoid doing checks everywhere. What I should change is the name to AssociateBLEDevice or add the compat sufix.